masterdrek/Flavor-Fusion
# Added Testing for `Modal.jsx`

# Add Tests for Modal Component to Improve Coverage

## Description

This pull request adds comprehensive tests for the `Modal` component to improve test coverage. The `Modal` component now has tests for various prop combinations and user interactions, ensuring all branches and states are covered.

### Changes Made

- Added tests for default props.
- Added tests for custom props with different combinations of `initialName`, `initialQuantity`, `showQuantity`, `isStep`, and `isNew`.
- Verified that the form submission behavior for both new and existing items is correct.
- Checked component's behavior when the cancel button is clicked.
- Improved overall test coverage to 100% for statements, functions, and lines, and 100% for branches.

### Test Coverage

***Before:***

![image](https://github.com/masterdrek/Flavor-Fusion/assets/91100832/527105fc-ecf6-46b6-a6c4-9fbab7617b52)


***After:***

![image](https://github.com/masterdrek/Flavor-Fusion/assets/91100832/c5f95d09-9767-4cf7-b80f-17728a371982)


### Checklist

- [x] Added tests to prove that our feature works.
- [x] Verified that all existing tests pass.
- [x] Verified that the code coverage has improved.

## How to Test

1. cd `react-frontend`
2. Run the tests with coverage: `npm test`
3. Verify that the test coverage is accurate 

## Related Issues

- Issue #49 


Please review this PR @Andrew76xx @masterdrek @tshellenberger @Baker-C
